File: cordapp-cpk2/src/test/resources/corda-api/src/main/java/net/corda/v5/persistence/MappedSchema.java
package net.corda.v5.persistence;

import org.jetbrains.annotations.NotNull;

public class MappedSchema {
    private final Class<?> schemaFamily;
    private final int version;
    private final Iterable<Class<?>> mappedTypes;

    public MappedSchema(@NotNull Class<?> schemaFamily, int version, @NotNull Iterable<Class<?>> mappedTypes) {
        this.schemaFamily = schemaFamily;
        this.version = version;
        this.mappedTypes = mappedTypes;
    }

    @NotNull
    public final String getName() {
        return schemaFamily.getName();
    }

    public final int getVersion() {
        return version;
    }

    @NotNull
    public final Iterable<Class<?>> getMappedTypes() {
        return mappedTypes;
    }
}

File: cordapp-cpk/src/test/resources/corda-api/src/main/kotlin/net/corda/v5/application/node/services/persistence/MappedSchema.kt
@file:Suppress("PackageDirectoryMismatch")
package net.corda.v5.application.node.services.persistence

open class MappedSchema(schemaFamily: Class<*>, val version: Int, val mappedTypes: Iterable<Class<*>>) {
    val name: String = schemaFamily.name
}
